internal/fs/handle/file.go

```diff
@@ -257,6 +261,8 @@ func (fh *FileHandle) tryEnsureReader(ctx context.Context, sequentialReadSizeMb
 	// If the inode is dirty, there's nothing we can do. Throw away our reader if
 	// we have one.
 	if !fh.inode.SourceGenerationIsAuthoritative() {
+		fh.mu.Lock()
+		defer fh.mu.Unlock()
 		if fh.reader != nil {
 			fh.reader.Destroy()
```

Comment:
Too many locks/unlocks in this code
can we avoid destroying the reader incase we need to read it from inode. that way we simplify the lock/unlock logic
Anyways reader will get destroyed when fileHandle is closed. 

@raj-prince - thoughts?